packages/next-on-pages/src/buildApplication/generateFunctionsMap.ts

```diff
@@ -144,37 +152,42 @@ async function processDirectoryRecursively(
 		setup.outputDir
 	);
 
-	await Promise.all(
-		functionFiles.map(async file => {
-			const filepath = join(dir, file);
-			if (await validateDir(filepath)) {
-				const dirResultsPromise = file.endsWith('.func')
-					? processFuncDirectory(setup, filepath)
-					: processDirectoryRecursively(setup, filepath);
-				const dirResults = await dirResultsPromise;
-				dirResults.invalidFunctions?.forEach(fn => invalidFunctions.add(fn));
-				dirResults.functionsMap?.forEach((value, key) =>
-					functionsMap.set(key, value)
-				);
-				dirResults.webpackChunks?.forEach((value, key) =>
-					webpackChunks.set(key, value)
-				);
-				dirResults.prerenderedRoutes?.forEach((value, key) =>
-					prerenderedRoutes.set(key, value)
-				);
-				dirResults.wasmIdentifiers?.forEach((value, key) =>
-					wasmIdentifiers.set(key, value)
-				);
-			}
-		})
-	);
+	// Note: the following could be implemented via `await Promise.all` since different directories are independent to each other
```

Comment:
A bit of a weird note to add here, but I am afraid that if I don't do so then at some point in the future someone (including me! 😁) could see this code, think of making it more efficient and swap it to the `await Promise.all` alternative

Doing so would break only large apps so the regression could even go unnoticed for a while and wreak havoc in the meantime 🤔

(PS: maybe we could somehow cover this with e2e tests? but that would require us to have some huge app or to potentially programmatically generate one on the spot? 🤔)

Reply:
No, this is a very good use of a comment!